src/components/TimelineNav/TimelineNav.tsx

```diff
@@ -51,18 +49,79 @@ export interface Props {
    * Slot for node to appear to the right of the title. Typically used to include a Badge, Button, or other component
    */
   titleAfter?: ReactNode;
-}
+};
 
-export interface TimelineNavItem {
-  props: {
-    'aria-label': string;
-    children: ReactNode;
-    completed?: boolean;
-    title?: string;
-    titleAfter?: ReactNode;
-    variant?: TimelineNavPanelVariant;
-  };
-}
+type TimelineNavPanelVariant =
+  | 'bullet'
+  | 'complete'
+  | 'error'
+  | 'incomplete'
+  | 'number'
+  | 'success'
+  | 'warning'
+  | undefined;
```

Comment:
This is a little funny; if a prop is set to `undefined` react will ignore it. Adding undefined to a type is a bit ambiguous. Instead it's better to use `?` for marking that.

I'm guessing this is inherited tho, so not requesting a change.

Reply:
Will change to `?` as using `undefined` is dumb 👍